From 9102afd0de7b724e1215a2a7515b2884aaab2926 Mon Sep 17 00:00:00 2001
From: yanhaodong <haodong.yan@spacemit.com>
Date: Mon, 23 Dec 2024 21:30:56 +0800
Subject: [PATCH 1387/1448] riscv : spacemit/k1/ZT001H: add charger jeita
 parameters

dts config does not fit charger driver,suspend failed:

Unable to handle kernel access to user memory without uaccess routines at virtual address 0000000000000000
 Oops [#2]
 Modules linked in:
 CPU: 4 PID: 173 Comm: ash Tainted: G      D            6.6.63+ #60
 Hardware name: ZT001H (DT)
 epc : 0x0
  ra : notifier_call_chain+0x62/0x108
 epc : 0000000000000000 ra : ffffffff80041784 sp : ffffffc801223c00
  gp : ffffffff832f33a0 tp : ffffffd90581b200 t0 : 0000000000000000
  t1 : 0000000000006000 t2 : 0000000000000000 s0 : ffffffc801223c70
  s1 : ffffffff8301e1a0 a0 : ffffffd9017d1a00 a1 : 0000000000000003
  a2 : 0000000000000000 a3 : 0000000000000000 a4 : 0000000000000000
  a5 : 0000000000000000 a6 : ffffffff83018e60 a7 : ffffffff83018e60
  s2 : 0000000000000000 s3 : 0000000000000000 s4 : fffffffffffffff8
  s5 : ffffffd9017d1a00 s6 : ffffffc801223c84 s7 : 0000000000000003
  s8 : 0000000000000000 s9 : 0000000000008000 s10: ffffffff832f9238
  s11: 0000000000000000 t3 : 0000000000000001 t4 : ffffffff832f9460
  t5 : 0000000000000001 t6 : 0000000000000000
 status: 0000000200000120 badaddr: 0000000000000000 cause: 000000000000000c
 Code: Unable to access instruction at 0xffffffffffffffec.
 ---[ end trace 0000000000000000 ]---

!!fix it!!

Change-Id: Idd982097ca79d362d1330a7ede73af909b6dc9bb
---
 arch/riscv/boot/dts/spacemit/k1-x_ZT001H.dts | 30 +++++++-------------
 1 file changed, 11 insertions(+), 19 deletions(-)

diff --git a/arch/riscv/boot/dts/spacemit/k1-x_ZT001H.dts b/arch/riscv/boot/dts/spacemit/k1-x_ZT001H.dts
index a16632b50e5e..ca02d98e7d14 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x_ZT001H.dts
+++ b/arch/riscv/boot/dts/spacemit/k1-x_ZT001H.dts
@@ -547,31 +547,23 @@ sgm4154x: sgm4154x@1a {
 		/* sw jeita  parameter customer defined*/
 		enable_sw_jeita;
 		jeita_temp_above_t4_cv = <0>;
-		jeita_temp_t3_to_t4_cv = <4100000>;
-		jeita_temp_t2_to_t3_cv = <4350000>;
-		jeita_temp_t1_to_t2_cv = <4350000>;
-		jeita_temp_t0_to_t1_cv = <0>;
-		jeita_temp_below_t0_cv = <0>;
+		jeita_temp_t3_to_t4_cv = <4400000>;
+		jeita_temp_t2_to_t3_cv = <4400000>;
+		jeita_temp_t1_to_t2_cv = <4400000>;
+		jeita_temp_below_t1_cv = <0>;
 
 		jeita_temp_above_t4_cc_current = <0>;
-		jeita_temp_t3_to_t4_cc_current = <1000000>;
-		jeita_temp_t2_to_t3_cc_current = <1250000>;
-		jeita_temp_t1_to_t2_cc_current = <1000000>;
-		jeita_temp_below_t0_cc_current = <0>;
+		jeita_temp_t3_to_t4_cc_current = <0000000>;
+		jeita_temp_t2_to_t3_cc_current = <2000000>;
+		jeita_temp_t1_to_t2_cc_current = <400000>;
+		jeita_temp_below_t1_cc_current = <0>;
 
-		temp_t4_thres = <50>;
-		temp_t4_thres_minus_x_degree = <48>;
+		temp_t4_thres = <60>;
 		temp_t3_thres = <45>;
-		temp_t3_thres_minus_x_degree = <45>;
-		temp_t2_thres = <20>;
-		temp_t2_thres_plus_x_degree = <15>;
+		temp_t2_thres = <15>;
 		temp_t1_thres = <0>;
-		temp_t1_thres_plus_x_degree = <0>;
-		temp_t0_thres = <0>;
-		temp_t0_thres_plus_x_degree = <2>;
-		temp_neg_10_thres = <0>;
 
-		sgm415xx-vchrg-uV = <4350000>;
+		sgm415xx-vchrg-uV = <4400000>;
 	};
 
 	cw2015: cw2015@62 {
-- 
2.47.0

